src/nextjs/middleware/withMiddlewareAuth.ts

```diff
@@ -119,12 +131,20 @@ export const withMiddlewareAuth: withMiddlewareAuth =
         );
       } else if (!authResult.user) {
         throw new Error('No auth user, redirecting');
+      } else if (
+        options.permissionGuard &&
+        !(await options.permissionGuard.isPermitted(authResult.user))
+      ) {
+        throw new NoPermissionError('User is not permitted, redirecting');
       }
 
       // Authentication successful, forward request to protected route
       return res;
     } catch (err: unknown) {
-      const { redirectTo = '/' } = options;
+      let { redirectTo = '/' } = options;
+      if (err instanceof NoPermissionError && !!options.permissionGuard) {
+        redirectTo = options.permissionGuard.redirectTo;
```

Comment:
I'd default to `redirectTo` from options if this is not defined.

Reply:
Yeah good idea. I've added that now